music21/musicxml/m21ToXml.py

```diff
@@ -1672,7 +1779,577 @@ def setScoreHeader(self):
+        MusicXML `<part>`:
+
+        >>> s = stream.Score()
+
+        Group 1: three staves.
+
+        >>> p1a = stream.PartStaff(id='p1a')
+        >>> p1a.insert(0, stream.Measure())
+        >>> p1b = stream.PartStaff(id='p1b')
+        >>> p1b.insert(0, stream.Measure())
+        >>> p1c = stream.PartStaff(id='p1c')
+        >>> p1c.insert(0, stream.Measure())
+        >>> sg1 = layout.StaffGroup([p1a, p1b, p1c])
+
+        Group 2: two staves.
+
+        >>> p2a = stream.PartStaff(id='p2a')
+        >>> p2a.insert(0, stream.Measure())
+        >>> p2b = stream.PartStaff(id='p2b')
+        >>> p2b.insert(0, stream.Measure())
+        >>> sg2 = layout.StaffGroup([p2a, p2b])
+
+        Group 3: one staff -- will not be merged.
+
+        >>> p3a = stream.PartStaff(id='p3a')
+        >>> p3a.insert(0, stream.Measure())
+        >>> sg3 = layout.StaffGroup([p3a])
+
+        Group 4: two staves, but no measures, will not be merged:
+
+        >>> p4a = stream.PartStaff(id='p4a')
+        >>> p4b = stream.PartStaff(id='p4b')
+        >>> sg4 = layout.StaffGroup([p4a, p4b])
+
+        Group 5: two staves, but no staff group
+
+        >>> p5a = stream.PartStaff(id='p5a')
+        >>> p5a.insert(0, stream.Measure())
+        >>> p5b = stream.PartStaff(id='p5b')
+        >>> p5b.insert(0, stream.Measure())
+
+        Group 6: same as Group 2, just to show that valid groups can come later
+
+        >>> p6a = stream.PartStaff(id='p6a')
+        >>> p6a.insert(0, stream.Measure())
+        >>> p6b = stream.PartStaff(id='p6b')
+        >>> p6b.insert(0, stream.Measure())
+        >>> sg6 = layout.StaffGroup([p6a, p6b])
+
+        Group 7: same as Group 6, but with Parts instead of PartStaffs
+
+        >>> p7a = stream.Part(id='p7a')
+        >>> p7a.insert(0, stream.Measure())
+        >>> p7b = stream.Part(id='p7b')
+        >>> p7b.insert(0, stream.Measure())
+        >>> sg7 = layout.StaffGroup([p7a, p7b])
+
+        >>> for el in (p1a, p1b, p1c, sg1, p2a, p2b, sg2, p3a, sg3,
+        ...            p4a, p4b, sg4, p5a, p5b, p6a, p6b, sg6, p7a, p7b, sg7):
+        ...     s.insert(0, el)
+
+        >>> SX = musicxml.m21ToXml.ScoreExporter(s)
+        >>> SX.joinableGroups()
+        [<music21.layout.StaffGroup <... p1a><... p1b><... p1c>>,
+         <music21.layout.StaffGroup <... p2a><... p2b>>,
+         <music21.layout.StaffGroup <... p6a><... p6b>>]
+        '''
+        staffGroups = self.stream.getElementsByClass('StaffGroup')
+        joinableGroups: List[StaffGroup] = []
+        # Joinable groups must consist of only PartStaffs with Measures
+        for sg in staffGroups:
+            if len(sg) <= 1:
+                continue
+            if not all(isinstance(p, stream.PartStaff) for p in sg):
+                continue
+            if not all(p.getElementsByClass('Measure') for p in sg):
+                continue
+            joinableGroups.append(sg)
+        return joinableGroups
```

Comment:
Now that this is factored out (awesome), what about using `yield` so that we don't store a list?